### PR TITLE
Readthedocs cleanup

### PR DESCRIPTION
**Pull request by mike-myers-tob**
- The information about running unit tests now points to the build guide for that information, since it varies slightly depending on platform and build tool
- All HTTP links are now HTTPS links (yay, security!)
- All references to this repo are osquery/osquery instead of facebook/osquery
- Minor format fixes and missing step added to build instructions
- Clarification added to how one is supposed to run a single test

### CODE REVIEW

**Comment by theopolis** [comment]
Curious what happened to the build here?

File: docs/wiki/deployment/debugging.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ osqueryi --config_path ./build/testing/invalid_osquery.conf --config_dump
 }
 ```
 
-In this example, I've added a C-style comment which [used](https://github.com/facebook/osquery/issues/1689) to be allowed in boost 1.58, but was deprecated and removed in 1.59. To be future-proof, stick to the JSON specification and do not include comments.
+In this example, I've added a C-style comment which [used](https://github.com/osquery/osquery/issues/1689) to be allowed in boost 1.58, but was deprecated and removed in 1.59. To be future-proof, stick to the JSON specification and do not include comments.
```
Comment (directionless) [inline docs/wiki/deployment/debugging.md]:
```suggestion
This example contains a C-style comment which was allowed in boost 1.58, but is deprecated and removed in 1.59. To be future-proof, stick to the JSON specification and do not include comments.
```

First person voice seems weird to me

File: docs/wiki/deployment/syslog.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
-osquery 1.7.3 introduced support for consuming and querying the Mac OSX system log via Apple System Log (ASL). osquery 1.7.4 introduced support for the Linux syslog via **rsyslog**. This document explains how to configure and use these syslog tables.
+osquery 1.7.3 introduced support for consuming and querying the macOS system log via Apple System Log (ASL). osquery 1.7.4 introduced support for the Linux syslog via **rsyslog**. This document explains how to configure and use these syslog tables.
 
 ## Apple macOS Syslog
 
-On Mac OSX, the `asl` virtual table makes use of Apple's ASL store, querying this structured store using the routines provided in [`asl.h`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/asl.3.html).
+On macOS, the `asl` virtual table makes use of Apple's ASL store, querying this structured store using the routines provided in [`asl.h`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/asl.3.html).
```
Comment (directionless) [inline docs/wiki/deployment/syslog.md]: Of course, we're probably going to remove that table.

File: docs/wiki/introduction/using-osqueryi.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## Executing SQL queries
```
Comment (directionless) [inline docs/wiki/introduction/using-osqueryi.md]: As long as you're in this file, I'd add a note about how `osqueryd -S` is osqueryi

**Comment by Smjert** [comment]
If you mean why it didn't start, it's because there's a conflict on building.md.

**Comment by mike-myers-tob** [comment]
> If you mean why it didn't start, it's because there's a conflict on building.md.

I hopefully resolved that conflict in this branch by incorporating the same changes you had made in Master. Now it should be mergeable again.

**Comment by Smjert** [comment]
Just to spare you of another rebase/merge, can I ask you to add `make` between the installed packages for Ubuntu in building.md?
When I removed the compilers I wrongly thought it would still pull it in.

Otherwise I can make a separate PR after this one.

**Pull request merged**